x-pack/plugins/security/server/lib/authorization/deep_freeze.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isObject } from 'lodash';
+
+export function deepFreeze(object) {
```

Comment:
At some point, we may want to consolidate this with the [OSS Version](https://github.com/kobelb/kibana/blob/login-deprecation-log/src/ui/public/metadata.js#L25), but no need to block this PR on that.

Reply:
Agreed, code-sharing is still a pain between OSS and x-pack, so I was postponing having to pull this into a `@kbn` package.